configure.ac

```diff
@@ -411,11 +412,19 @@ if test "$CFG_ENABLE_COVERAGE" = "yes"; then
   #  _MY_CXX_CHECK_OPT(CXX,-fkeep-inline-functions)
   # Otherwise static may not show as uncovered
   _MY_CXX_CHECK_OPT(CXX,-fkeep-static-functions)
-  # Exceptions can pollute the branch coverage data
-  _MY_CXX_CHECK_OPT(CXX,-fno-exceptions)
-  # Define-out some impossible stuff
+  # Similarly for inline functions. - This is too slow. See Makefile_obj instead.
+  #_MY_CXX_CHECK_OPT(CXX,-fkeep-inline-functions)
+  # Make sure profiling is thread-safe
+  _MY_CXX_CHECK_OPT(CXX,-fprofile-update=atomic)
+  # Ensure data files can be written from parallel runs
+  _MY_CXX_CHECK_OPT(CXX,-fprofile-reproducible=parallel-runs)
+  # Save source files as absolute paths in gcno files
```

Comment:
FWIW these 2 options are what allows us to keep it simple. It enables updating the .gcda files from multiple invocations, and from multiple threads safely.